py/server/tests/test_table_factory.py

```diff
@@ -200,11 +200,76 @@ def test_dtw_single_string_arg(self):
 
         col_defs = {"A_Long": dtypes.long}
         table_writer = DynamicTableWriter(col_defs)
-        table_writer.write_row(10**10)
+        table_writer.write_row(10 ** 10)
         t = table_writer.table
         self.wait_ticking_table_update(t, row_count=1, timeout=5)
         self.assertIn("10000000000", t.to_string())
 
+    def test_input_table(self):
+        col_defs = {
+            "ByteArray": dtypes.byte_array,
+            "ShortArray": dtypes.short_array,
+            "Int32Array": dtypes.int32_array,
+            "LongArray": dtypes.long_array,
+            "Float32Array": dtypes.float32_array,
+            "DoubleArray": dtypes.double_array,
+            "StringArray": dtypes.string_array,
+        }
+        with DynamicTableWriter(col_defs) as table_writer:
+            b_array = dtypes.array(dtypes.byte, [1, 1, 1])
+            s_array = dtypes.array(dtypes.short, [128, 228, 328])
+            i_array = dtypes.array(dtypes.int32, [32768, 42768, 52768])
+            l_array = dtypes.array(dtypes.long, [2 ** 32, 2 ** 33, 2 ** 36])
+            f_array = dtypes.array(dtypes.float32, [1.0, 1.1, 1.2])
+            d_array = dtypes.array(dtypes.double, [1.0 / 2 ** 32, 1.1 / 2 ** 33, 1.2 / 2 ** 36])
+            str_array = dtypes.array(dtypes.string, ["some", "not so random", "text"])
+            table_writer.write_row(b_array, s_array, i_array, l_array, f_array, d_array, str_array
+                                   )
+        t = table_writer.table
+        self.wait_ticking_table_update(t, row_count=1, timeout=5)
```

Comment:
For clarity, I think this table that we are using as input to the inputs tables should be created via new_table instead of table writer.